src/openfluid/ui/waresdev/FragmentsImportWorker.cpp

```diff
@@ -145,28 +145,25 @@ bool FragmentsImportWorker::importElement(const QString& GitUrl, const QString&
         {RootPath.toStdString(), DestSubPath.toGeneric()});
 
       // removing also the reference in .git folder
-      openfluid::tools::FilesystemPath GitSubmoduleTargetDirectory = openfluid::tools::FilesystemPath(
-        {RootPath.toStdString(), ".git", "modules", DestSubPath.toGeneric()});
-      if (GitSubmoduleTargetDirectory.isDirectory())
+      if (openfluid::ui::waresdev::cleanModuleDir(RootPath, DestSubPath) == 0)
       {
-        GitSubmoduleTargetDirectory.removeDirectory();
         emit info(tr("Fragment directory removed from .git/modules since import not successful"));
       }
+      
```

Comment:
I miss some information to clearly understand this part (current and following lines)